File: www.spoj.com/42:Adding_Reversed_Numbers/42.cpp
/*
 * 42.cpp
 *
 *  Created on: Dec 15, 2016
 *      Author: frankt
 */

#include <iostream>
#include <string>
#include <stdint.h>

using namespace std;

uint32_t Reverse(uint32_t number)
{
    uint32_t result = 0;
    while(number)
    {
        result = result*10 + number%10;
        number /= 10;
    }
    return result;
}

int main()
{
    int32_t case_count;
    cin>>case_count;
    while(case_count--)
    {
        uint32_t a;
        uint32_t b;
        cin>>a>>b;
        cout<<Reverse(Reverse(a) + Reverse(b))<<"\n";
    }
    return 0;
}

File: www.spoj.com/42:Adding_Reversed_Numbers/42.go
package main

import (
	"fmt"
)

func reverse(number int) int {
	var result int = 0
	for number != 0 {
		result = result * 10 + number % 10
		number /= 10
	}
	return result
}

func main() {
	var caseCount int
	fmt.Scan(&caseCount)
	for i := 0; i < caseCount; i++ {
		var a, b int
		fmt.Scan(&a, &b)
		fmt.Println(reverse(reverse(a) + reverse(b)))
	}
}
